SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -46,7 +46,7 @@ def fields(self, m):
             # Compute the linear operation without forming the full dense G
             fields = mkvc(self.linear_operator())
         else:
-            fields = self.G @ (self.rho).astype(np.float32)
+            fields = self.G @ (self.rho).astype(self.sensitivity_dtype)
```

Comment:
@jcapriot, after checking [`astype` docs](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html), I think the method by default copies the array even if it has the same type as the given one. I think we should enforce not to copy the array with `copy=False`.
```suggestion
            fields = self.G @ (self.rho).astype(self.sensitivity_dtype, copy=False)
```

I've just tested its behaviour with this little snippet:
```python
import numpy as np
a = np.array([1, 2, 3], dtype=np.float64)
b = a.astype(np.float64)
c = a.astype(np.float64, copy=False)
```
```python
a is b
```
```
False
```
```python
a is c
```
```
True
```

Reply:
Ah yes, thanks